examples/v2/project_creation/project.py

```diff
@@ -147,6 +146,44 @@ def GenerateConfig(context):
             'dependsOn': action_dependency
         }
     })
+  if context.properties.get('shared_vpc_host'):
+     resources.append({
+        'name': project_id + '-xpn-host',
+        'type': 'compute.beta.xpnHost',
+        'properties': {
+            'organization-id': context.properties['organization-id'],
+            'billing-account-name': context.properties['billing-account-name'],
+            'project': project_id,
+        },
+        'metadata': {
+            'dependsOn': [
+                ApiResourceName(project_id, 'compute.googleapis.com'),
+                project_id,
+            ],
+        }
+     })
+  if context.properties.get('shared_vpc_service_of'):
+      resources.append({
+        'name': project_id + '-xpn-service-' +
+            context.properties['shared_vpc_service_of'],
+        'type': 'compute.beta.xpnResource',
```

Comment:
Can you use
action: gcp-types/compute-beta:compute.projects.enableXpnHost
instead of type?

The difference is that when you delete the deployment compute.beta.xpnResource will make a call to XPNHost, but the action only will be called during the create of the deployment

Reply:
I tried, and it's not possible without adding a lot more stuff:
* Creating a XPNHost resource, whether by `gcp-types/compute-beta:compute.projects.enableXpnHost` or `compute.beta.xpnHost`, also creates a lien that prevents the removal of the host project. However, if you use `gcp-types/compute-beta:compute.projects.enableXpnHost`, the lien is not deleted when deleting the deployment, leading to an error as the project itself will still be protected against deletion.
* Same kind of thing for XPNResource: using `gcp-types/compute-beta:compute.projects.enableXpnHost` breaks the dependency graph between the resources upon deletion. When deleting the deployment, does not know that the service project must be detached before removing the `xpnHost` resource, leading to the following error:

     Cannot disable project as an XPN host because it has active service projects.

Keeping this as it is right now is much simpler because DM deals by itself with the order in which stuff have to be created and deleted.